src/components/common/API.jsx

```diff
@@ -1146,6 +1146,40 @@ class API {
     getColors() {
         return this.colors;
     }
+
+    /**
+     * Vendor
+     */
+
+    async getGProfilerLink(
+        clusteringID,
+        clusterID,
+        topNumFeatures,
+        organism,
+        token
+    ) {
+        let query = {
+            loomFilePath: this.getActiveLoom(),
+            clusteringID: clusteringID,
+            clusterID: clusterID,
+            topNumFeatures: topNumFeatures,
+            organism: token !== null ? token : organism,
+        };
+        console.log('Run getGProfilerLink');
+        console.log(query);
+
+        return new Promise((resolve, reject) => {
+            this.getConnection().then((gbc) => {
+                if (DEBUG) console.log('getGProfilerLink', query);
+                gbc.services.scope.Main.getGProfilerLink(
```

Comment:
> Ah, here's where the link is created on the server? Why are you using a raw Promise in an `async` function?

I understand but then how would the Promise resolve since the result is nested ?

> Why not create the link here? Do you not have the information you need?

Data is there but processing the data is much easier too me on the back-end in Python (could be biased)

Reply:
If we did decide to do this fully on the frontend, the ReactTable module could be used to filter the data, and then this data could be extracted (https://spectrum.chat/react-table/general/best-way-to-get-currently-visible-data~3bc65318-678d-4f7e-88f5-7c5050668287). In this case, we don't need to do anything to deal with the data as we just get the gene list directly, people would also have a better refinement on which genes they are submitting.

Although this would mean submitting multiple sets (as is done now) would be difficult.

Reply:
See comment above

Reply:
I have a feeling that it's going to be harder in JS but I can try

Reply:
Sure it might be harder. Let me know if you want to pair.